### PR TITLE
Groups auth

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

https://github.com/dust-tt/dust/issues/6448
Implements auth.groups()

## Risk

Can break auth 😬 
Can be rolled back. 

## Deploy Plan

Deploy front.

### CODE REVIEW

File: front/lib/resources/group_resource.ts [hunk 1]
```diff
@@ -199,9 +243,64 @@ export class GroupResource extends BaseResource<GroupModel> {
...
   }
 
+  static async fetchActiveGroupsOfUserInWorkspace({
+    user,
+    workspace,
+    transaction,
+  }: {
+    user: UserResource;
+    workspace: LightWorkspaceType;
+    transaction?: Transaction;
```
Comment (spolu) [inline front/lib/resources/group_resource.ts:253]: let's not have a transaction arg if we don't need it :pray:

File: front/lib/resources/group_resource.ts [hunk 2]
```diff
@@ -199,9 +243,64 @@ export class GroupResource extends BaseResource<GroupModel> {
...
+
+    const groups = [globalGroup, ...regularGroups];
+
+    return groups.map((group) => new this(GroupModel, group.get()));
+  }
+
   async addMember(
     auth: Authenticator,
-    userId: string,
+    user: UserType,
```
Comment (PopDaph) [inline front/lib/resources/group_resource.ts:303]: This is feedback from my last PR, implemented here 🙈

File: front/lib/resources/group_resource.ts [hunk 3]
```diff
@@ -116,37 +123,58 @@ export class GroupResource extends BaseResource<GroupModel> {
...
     });
 
     return groups.map((group) => new this(GroupModel, group.get()));
   }
 
-  static async fetchWorkspaceSystemGroup(
-    auth: Authenticator,
+  static async fetchWorkspaceGroupsFromKey(
+    key: KeyResource,
+    transaction?: Transaction
```
Comment (spolu) [inline front/lib/resources/group_resource.ts:147]: let remove transaction param if we don't need it

File: front/lib/resources/group_resource.ts [hunk 4]
```diff
@@ -172,6 +200,22 @@ export class GroupResource extends BaseResource<GroupModel> {
     return new this(GroupModel, group.get());
   }
 
+  static async fetchWorkspaceGroups(
+    auth: Authenticator,
+    transaction?: Transaction
```
Comment (spolu) [inline front/lib/resources/group_resource.ts:205]: let's remove transaction param if we don't need it

File: front/lib/resources/group_resource.ts [hunk 5]
```diff
@@ -116,37 +123,58 @@ export class GroupResource extends BaseResource<GroupModel> {
...
     return groups.map((group) => new this(GroupModel, group.get()));
   }
 
-  static async fetchWorkspaceSystemGroup(
-    auth: Authenticator,
+  static async fetchWorkspaceGroupsFromKey(
+    key: KeyResource,
+    transaction?: Transaction
+  ): Promise<GroupResource[]> {
+    const group = await this.model.findOne({
```
Comment (spolu) [inline front/lib/resources/group_resource.ts:149]: Let's leave a `// TODO(GROUPS_INFRA)` saying that we need to pull the groups associated with the key once that's built

File: front/lib/auth.ts [hunk 6]
```diff
@@ -346,10 +367,12 @@ export class Authenticator {
       throw new Error(`Could not find workspace with sId ${workspaceId}`);
     }
 
+    let globalGroup: GroupResource | null = null;
```
Comment (spolu) [inline front/lib/auth.ts:370]: `internalBuilderForWorkspace` will likely have to go away or change as some of its use assume ability to operate on any group I believe?

In any case not for now but maybe leave a `// TODO(GROUPS_INFRA)` that this is not group compatible anymore and will have to be refactored?

File: front/lib/auth.ts [hunk 7]
```diff
@@ -383,10 +407,12 @@ export class Authenticator {
...
     let subscription: SubscriptionType | null = null;
     let flags: WhitelistableFeature[] = [];
 
-    [subscription, flags] = await Promise.all([
+    [globalGroup, subscription, flags] = await Promise.all([
+      GroupResource.internalFetchWorkspaceGlobalGroup(workspace.id),
```
Comment (spolu) [inline front/lib/auth.ts:415]: I think internalAdminForWorkspace really assume access to everything no? Should really be internalSystemForWorkspace which gives a system API. Let's file some tasks to not forget to figure that all out.

**Review by spolu — APPROVED** [review]
LGTM

Left a few comments

File: front/lib/auth.ts (hunk 7 above)
Comment (PopDaph) [reply front/lib/auth.ts:418]: I added a comment and I'll add a task. 👍🏻

**Pull request merged**